src/java.net.http/share/classes/jdk/internal/net/http/common/HttpBodySubscriberWrapper.java

```diff
@@ -157,7 +178,7 @@ public void onSubscribe(Flow.Subscription subscription) {
     public void onNext(List<ByteBuffer> item) {
         if (completed.get()) {
             if (subscription != null) {
-                subscription.cancel();
+                subscription.subscription.cancel();
```

Comment:
Are we intentionally accessing and cancelling the underlying subscription here, instead of the wrapper? Do we intentionally want to bypass the on-cancellation logic in this flow?

Reply:
Thank you for that explanation. Looks good to me.